src/main/kotlin/it/pagopa/ecommerce/scheduler/queues/TransactionAuthRequestedEventsConsumer.kt

```diff
@@ -1,25 +1,48 @@
 package it.pagopa.ecommerce.scheduler.queues
 
+import com.azure.core.util.BinaryData
 import com.azure.spring.messaging.AzureHeaders
 import com.azure.spring.messaging.checkpoint.Checkpointer
+import it.pagopa.ecommerce.scheduler.repositories.TransactionsEventStoreRepository
+import it.pagopa.ecommerce.scheduler.services.NodeService
+import it.pagopa.ecommerce.scheduler.services.RefundService
+import it.pagopa.generated.ecommerce.nodo.v1.dto.ClosePaymentRequestV2Dto
+import it.pagopa.transactions.documents.TransactionAuthorizationRequestedEvent
+import it.pagopa.transactions.documents.TransactionAuthorizationStatusUpdateData
+import it.pagopa.transactions.utils.TransactionEventCode
+import kotlinx.coroutines.reactor.mono
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.integration.annotation.ServiceActivator
 import org.springframework.messaging.handler.annotation.Header
 import org.springframework.messaging.handler.annotation.Payload
 import org.springframework.stereotype.Service
+import java.util.*
 
 @Service
-class TransactionAuthRequestedEventsConsumer {
+class TransactionAuthRequestedEventsConsumer(
+    @Autowired private val nodeService: NodeService,
+    @Autowired private val refundService: RefundService,
+    @Autowired private val transactionsAuthorizationStatusUpdateEventRepository: TransactionsEventStoreRepository<TransactionAuthorizationStatusUpdateData>,
+) {
 
     var logger: Logger = LoggerFactory.getLogger(TransactionAuthRequestedEventsConsumer::class.java)
-    
+
     @ServiceActivator(inputChannel = "transactionauthrequestedchannel")
     fun messageReceiver(@Payload payload: ByteArray, @Header(AzureHeaders.CHECKPOINTER) checkpointer: Checkpointer) {
-        val message = String(payload)
-        checkpointer.success().doOnSuccess({
-            logger.info("Message '{}' successfully checkpointed", message)
-        })
-        .block()
+        checkpointer.success().block()
+        logger.info("Message '{}' successfully checkpointed", String(payload))
+
+        val authorizationRequestedEvent = BinaryData.fromBytes(payload).toObject(TransactionAuthorizationRequestedEvent::class.java)
+        val transactionId = authorizationRequestedEvent.transactionId
+
+        val statusUpdateEvent = transactionsAuthorizationStatusUpdateEventRepository.findByTransactionIdAndEventCode(transactionId, TransactionEventCode.TRANSACTION_AUTHORIZATION_STATUS_UPDATED_EVENT).block()
+
+        if (statusUpdateEvent == null) {
+            logger.info("Refunding transaction ${authorizationRequestedEvent.transactionId}")
+            refundService.requestRefund(authorizationRequestedEvent.data.requestId).block()
```

Comment:
the retry logic in case of error we handle in an adhoc PR, but can you add a log related to this exception? just like TODO.